appgate/resource_appgate_device_script.go

```diff
@@ -0,0 +1,229 @@
+package appgate
+
+import (
+	"bufio"
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"time"
+
+	"github.com/appgate/terraform-provider-appgate/client/v12/openapi"
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAppgateDeviceScript() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAppgateDeviceScriptCreate,
+		Read:   resourceAppgateDeviceScriptRead,
+		Update: resourceAppgateDeviceScriptUpdate,
+		Delete: resourceAppgateDeviceScriptDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+
+			"device_script_id": {
+				Type:        schema.TypeString,
+				Description: "ID of the object.",
+				Computed:    true,
+			},
+
+			"name": {
+				Type:        schema.TypeString,
+				Description: "Name of the object.",
+				Required:    true,
+			},
+
+			"notes": {
+				Type:        schema.TypeString,
+				Description: "Notes for the object. Used for documentation purposes.",
+				Default:     DefaultDescription,
+				Optional:    true,
+			},
+
+			"tags": {
+				Type:        schema.TypeSet,
+				Description: "Array of tags.",
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+
+			"filename": {
+				Type:        schema.TypeString,
+				Description: "The name of the file to be downloaded as to the client devices.",
+				Required:    true,
+			},
+
+			"file": {
+				Type:          schema.TypeString,
+				Description:   "Path to the Device Script binary.",
+				Optional:      true,
+				ConflictsWith: []string{"content"},
+			},
+
+			"content": {
+				Type:          schema.TypeString,
+				Description:   "The Device Script content.",
+				Optional:      true,
+				ConflictsWith: []string{"file"},
+			},
```

Comment:
If I understand well, this resource allows storing the device script either directly as a string or as a file for arbitrary binary content. Both are mapped to a base64 string in the controller API right?

Reply:
yes exactly, just for ease of use for the developers so you can either
```hcl
resource "appgate_device_script" "example_device_script" {
  name     = "device_script_name"
  filename = "script.sh"
  content  = <<-EOF
#!/usr/bin/env bash
echo "hello world"
EOF
}
```
or
```hcl
resource "appgate_device_script" "example_device_script" {
  name     = "device_script_name"
  filename = "script.sh"
  file     = base64encode(file("${path.module}/hello.txt"))
}
```

Reply:
Ok, that's nice. Can you add the file example to the docs?
Feel free to merge after.